src/lib/sql/sql_translator.cpp

```diff
@@ -683,34 +683,34 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_cross_product(const st
 }
 
 void SQLTranslator::_translate_select_list_groupby_having(const hsql::SelectStatement& select) {
-  const auto& input_expressions = _current_lqp->column_expressions();
-
-  auto pre_aggregate_expression_set = ExpressionUnorderedSet{input_expressions.begin(), input_expressions.end()};
-  auto pre_aggregate_expressions =
-      std::vector<std::shared_ptr<AbstractExpression>>{input_expressions.begin(), input_expressions.end()};
+  auto pre_aggregate_expression_set = ExpressionUnorderedSet{};
+  auto pre_aggregate_expressions = std::vector<std::shared_ptr<AbstractExpression>>{};
   auto aggregate_expression_set = ExpressionUnorderedSet{};
   auto aggregate_expressions = std::vector<std::shared_ptr<AbstractExpression>>{};
 
   // Visitor that identifies aggregates and their arguments
   const auto find_aggregates_and_arguments = [&](auto& sub_expression) {
-    if (sub_expression->type != ExpressionType::Aggregate) return ExpressionVisitation::VisitArguments;
-
     /**
      * If the AggregateExpression has already been computed in a previous node (consider "x" in
      * "SELECT x FROM (SELECT MIN(a) as x FROM t) AS y)", it doesn't count as a new Aggregate and is therefore not
      * considered an "Aggregate" in the current SELECT list. Handling this as a special case seems hacky to me as well,
      * but it's the best solution I can come up with right now.
      */
-    if (_current_lqp->find_column_id(*sub_expression)) return ExpressionVisitation::DoNotVisitArguments;
+    if (_current_lqp->find_column_id(*sub_expression)) {
+      if (pre_aggregate_expression_set.emplace(sub_expression).second) {
```

Comment:
Can you remember why 

```c++

      if (pre_aggregate_expression_set.emplace(sub_expression).second) {
        pre_aggregate_expressions.emplace_back(sub_expression);
      }
```

was necessary here? I removed it and apart from affecting the order of columns in some non-root LQP nodes, it didn't break anything.

Also, the lambda is called "find_aggregates_and_arguments" and this block feels alien here as it doesn't work with aggregates at all.

Reply:
Hmm, no, I can't find a way to break it even without that block. Seems logical as well, because if we don't add it here, it's not part of an aggregate. If it's not part of an aggregate, it has to be part of the `GROUP BY`, and will get added later.